src/main/java/com/microsoft/graph/serializer/EdmNativeTypeSerializer.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.graph.serializer;
+
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.util.UUID;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.microsoft.graph.logger.ILogger;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/** Deserializer for native EDM types from the service. Used for actions and functions that return native types for example. */
+public class EdmNativeTypeSerializer {
+    /**
+     * Deserializes native EDM types from the service. Used for actions and functions that return native types for example.
+     * @param <T> Expected return type.
+     * @param json json to deserialize.
+     * @param type class of the expected return type.
+     * @param logger logger to use.
+     * @return the deserialized type or null.
+     */
+    @Nullable
+    public static <T> T deserialize(@Nonnull final JsonElement json, @Nonnull final Class<T> type, @Nonnull final ILogger logger) {
+        if (json == null || type == null) {
+			return null;
+        } else if(json.isJsonPrimitive()) {
+            return getPrimitiveValue(json, type);
+		} else if(json.isJsonObject()) {
+            final JsonElement element = json.getAsJsonObject().get("@odata.null");
+            if(element != null && element.isJsonPrimitive()) {
+                return getPrimitiveValue(element, type);
+            } else {
+                return null;
+            }
+        }
+        return null;
+    }
+    @SuppressWarnings("unchecked")
+    private static <T> T getPrimitiveValue(final JsonElement json, final Class<T> type) {
+        if(type == Boolean.class) {
+            return (T) Boolean.valueOf(json.getAsBoolean());
+        } else if(type == String.class) {
```

Comment:
Edm.Binary, Edm.Date, Edm.DateTimeOffset, Edm.Duration,  Edm.Guid, and Edm.TimeOfDay will be treated as strings, correct?

http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_PrimitiveValue

Reply:
Edm Guid is below (UUID in Java). Binary, Date, DateTimeOffset, Duration and TimeOfDay have their dedicated serializers.